jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -237,49 +287,76 @@ private static DiagnosticProperties doHelp(String diagnosticCommand) {
 			+ " Options: -all : include all objects, including dead objects%n"
 			+ "NOTE: this utility may significantly affect the performance of the target VM.%n";
-	
+
 	@SuppressWarnings("nls")
 	private static final String DIAGNOSTICS_GC_RUN_HELP = "Run the garbage collector.%n"
-			+ " Format: " + DIAGNOSTICS_GC_RUN + "%n"
+			+ FORMAT_PREFIX + DIAGNOSTICS_GC_RUN + "%n"
 			+ "NOTE: this utility may significantly affect the performance of the target VM.%n";
-	
+
 	@SuppressWarnings("nls")
 	private static final String DIAGNOSTICS_THREAD_PRINT_HELP = "List thread information.%n"
-			+ " Format: " + DIAGNOSTICS_THREAD_PRINT + " [options]%n"
+			+ FORMAT_PREFIX + DIAGNOSTICS_THREAD_PRINT + " [options]%n"
 			+ " Options: -l : print information about ownable synchronizers%n";
-	
+
+	private static final String DIAGNOSTICS_DUMP_HEAP_HELP = "Create a heap dump.%n" //$NON-NLS-1$
+			+ FORMAT_PREFIX + DIAGNOSTICS_DUMP_HEAP + FILE_PATH_HELP
+			+ DIAGNOSTICS_GC_HEAP_DUMP + " is an alias for " + DIAGNOSTICS_DUMP_HEAP; //$NON-NLS-1$
+
+	private static final String DIAGNOSTICS_DUMP_JAVA_HELP = "Create a javacore file.%n" //$NON-NLS-1$
+			+ FORMAT_PREFIX + DIAGNOSTICS_DUMP_JAVA + FILE_PATH_HELP;
+
+	private static final String DIAGNOSTICS_DUMP_SNAP_HELP = "Dump the snap trace buffer.%n" //$NON-NLS-1$
+			+ FORMAT_PREFIX + DIAGNOSTICS_DUMP_SNAP + FILE_PATH_HELP;
+
+	private static final String DIAGNOSTICS_DUMP_SYSTEM_HELP = "Create a native core file.%n" //$NON-NLS-1$
+			+ FORMAT_PREFIX + DIAGNOSTICS_DUMP_SYSTEM + FILE_PATH_HELP;
+
 	/* Initialize the command and help text tables */
 	static {
 		commandTable = new HashMap<>();
 		helpTable = new HashMap<>();
-		
+
 		commandTable.put(DIAGNOSTICS_HELP, DiagnosticUtils::doHelp);
 		helpTable.put(DIAGNOSTICS_HELP, DIAGNOSTICS_HELP_HELP);
-		
+
 		commandTable.put(DIAGNOSTICS_GC_CLASS_HISTOGRAM, DiagnosticUtils::getHeapStatistics);
 		helpTable.put(DIAGNOSTICS_GC_CLASS_HISTOGRAM, DIAGNOSTICS_GC_CLASS_HISTOGRAM_HELP);
-		
+
 		commandTable.put(DIAGNOSTICS_GC_RUN, s -> runGC());
 		helpTable.put(DIAGNOSTICS_GC_RUN, DIAGNOSTICS_GC_RUN_HELP);
-		
+
 		commandTable.put(DIAGNOSTICS_THREAD_PRINT, DiagnosticUtils::getThreadInfo);
 		helpTable.put(DIAGNOSTICS_THREAD_PRINT, DIAGNOSTICS_THREAD_PRINT_HELP);
-	}
 
+		commandTable.put(DIAGNOSTICS_DUMP_HEAP, DiagnosticUtils::doDump);
+		helpTable.put(DIAGNOSTICS_DUMP_HEAP, DIAGNOSTICS_DUMP_HEAP_HELP);
+
+		commandTable.put(DIAGNOSTICS_GC_HEAP_DUMP, DiagnosticUtils::doDump);
+		helpTable.put(DIAGNOSTICS_GC_HEAP_DUMP, DIAGNOSTICS_DUMP_HEAP_HELP);
+
+		commandTable.put(DIAGNOSTICS_DUMP_JAVA, DiagnosticUtils::doDump);
+		helpTable.put(DIAGNOSTICS_DUMP_JAVA, DIAGNOSTICS_DUMP_JAVA_HELP);
+
+		commandTable.put(DIAGNOSTICS_DUMP_SNAP, DiagnosticUtils::doDump);
+		helpTable.put(DIAGNOSTICS_DUMP_SNAP, DIAGNOSTICS_DUMP_SNAP_HELP);
+
+		commandTable.put(DIAGNOSTICS_DUMP_SYSTEM, DiagnosticUtils::doDump);
+		helpTable.put(DIAGNOSTICS_DUMP_SYSTEM, DIAGNOSTICS_DUMP_SYSTEM_HELP);
+	}
```

Comment:
These two `HashMap` can be lazy initialized to help startup performance.

Reply:
This class is not loaded unless there is a diagnostic operation, so startup will not be affected.